packages/did/src/Did.chain.ts

```diff
@@ -380,8 +345,11 @@ export async function queryDeletedDidIdentifiers(): Promise<DidIdentifier[]> {
 
 // ### EXTRINSICS types
 
-export type PublicKeyEnum = Record<string, Uint8Array>
-export type SignatureEnum = Record<string, Uint8Array>
+export type PublicKeyEnum<K extends NewDidKey> = JsonEnum<
+  Capitalize<K['type']>,
```

Comment:
I see how they may seem related, but what you are referring to is a union of string literals, while this is a utility type that helps creating these enum types like `{ Ed25519: '0x123' }` that polkadot requires. It's quite a pain to get from the former to the latter, this is why this exists.

Reply:
I feel this would work while also saving us from the future pain of trying to find ends in the type transformations labyrinth:
```ts
export type PublicKeyEnum<Type extends MultiSignature['type']> = JsonEnum<
  Type,
  Uint8Array
>
export type SignatureEnum = PublicKeyEnum<MultiSignature['type']>